tests/test_echo.py

```diff
@@ -200,7 +201,7 @@ async def test_purerpc_client_disconnect(echo_pb2, echo_grpc):
         port = await tg.start(server.serve_async)
 
         # client
-        with pytest.raises(anyio.ClosedResourceError):
+        with pytest.raises(anyio.ClosedResourceError), unwrap_exceptiongroups_single():
             async with purerpc.insecure_channel("localhost", port) as channel:
                 stub = echo_grpc.EchoStub(channel)
```

Comment:
Especially here.  Now the purerpc API itself may raise ExceptionGroup.  I don't think this implementation detail should be exposed, and the API should continue to raise `ClosedResourceError`.

> I explained it in a thread of another project, but merely patching tests to handle strict exception groups doesn't provide any assurance that the runtime code supports strict exceptions. For example, where the API was previously raising a bare exception, it might now raise an ExceptionGroup (the API should probably never raise ExceptionGroup, since internal use of nurseries/concurrency is an implementation detail). Likewise, internally to the runtime, if there are any except statements catching bare exceptions, they need to be ported to excptiongroup catches.